mula/scheduler/app.py

```diff
@@ -252,8 +252,65 @@ def run(self) -> None:
             interval=self.ctx.config.monitor_organisations_interval,
         )
 
+        # Start metrics collecting
+        self.run_in_thread(
+            name="metrics_collector",
+            func=self.collect_metrics,
+            interval=1,
+        )
+
         # Main thread
         while not self.stop_event.is_set():
             time.sleep(0.01)
 
         self.shutdown()
+
+    def collect_metrics(self) -> None:
```

Comment:
For my understanding: the only advantage of running this in a thread vs. on every request is some performance benefits when we get multiple requests per second? A Gauge is "forgetful" right?

Reply:
I went back on forward on this. I decided eventually to implement it this way because otherwise (if I remember correctly) application-wide metrics meant that I had to pass the app as an argument to the server in order to get those.